torch/distributed/_shard/sharded_tensor/_ops/_common.py

```diff
@@ -4,6 +4,56 @@
     Shard,
     ShardedTensor,
 )
+from torch.distributed._shard.partial_tensor import _PartialTensor
+from torch.distributed._shard.replicated_tensor import ReplicatedTensor
+from torch.utils._pytree import tree_map
+
+def _basic_validation(op, args=(), kwargs=None):
+    """
+    Common validation across all ops go in here.
+    """
+    if len(args) == 0 and (kwargs is None or len(kwargs) == 0):
+        raise ValueError(f" No input for '{op.__name__}'!")
+
+    # Validate types
+    has_distributed_tensor = False
+
+    def is_distributed_tensor(e):
+        nonlocal has_distributed_tensor
+        if isinstance(e, ReplicatedTensor) or isinstance(e, _PartialTensor) or isinstance(e, ShardedTensor):
+            has_distributed_tensor = True
+
+    tree_map(is_distributed_tensor, args)
+    tree_map(is_distributed_tensor, kwargs)
+
+    if not has_distributed_tensor:
+        raise TypeError(
+            f"torch function '{op.__name__}', with args: {args} and "
+            f"kwargs: {kwargs} are called without any DistributedTensor!"
+        )
+
+    # Validate all DistributedTensors use the same PG.
+    cur_pg = None
+
+    def validate_pg(e):
+        nonlocal cur_pg
+        if isinstance(e, ReplicatedTensor) or isinstance(e, _PartialTensor):
+            if cur_pg is not None and e.process_group is not cur_pg:
+                raise RuntimeError(
+                    'All distributed tensors should use the '
+                    'same ProcessGroup if used together in an op.'
+                )
+            cur_pg = e.process_group
+        elif isinstance(e, ShardedTensor):
+            if cur_pg is not None and e._process_group is not cur_pg:
```

Comment:
these two are essentially the same logic except that `ShardedTensor` uses `_process_group` but the other two uses `process_group`? Maybe we can consolidate this later in a separate PR to all use `_process_group`?